install/installer/pkg/common/common.go

```diff
@@ -350,11 +350,26 @@ func RepoName(repo, name string) string {
 	if err != nil {
 		panic(fmt.Sprintf("cannot parse image repo %s: %v", ref, err))
 	}
-	return pref.String()
+
+	prefString := pref.String()
+
+	if cfg.Repository != GitpodContainerRegistry {
+		// If not in the Gitpod registry, don't use namespaces unless specified in the config.
+		// This is to match the KOTS registry format of no namespace in the name
+		s := strings.Split(prefString, "/")
+
+		noNamespace := []string{
+			cfg.Repository,
+			s[len(s)-1],
+		}
+		return strings.Join(noNamespace, "/")
+	}
+
+	return prefString
 }
 
-func ImageName(repo, name, tag string) string {
-	ref := fmt.Sprintf("%s:%s", RepoName(repo, name), tag)
+func ImageName(repo, name, tag string, cfg *config.Config) string {
```

Comment:
nit question: why is `cfg` param added considering that only `cfg.Repository` is used, which in all calls  to the function seem to be same as `repo`?

Reply:
Fair question. It's done more as a nod to future-proofing the function - whilst I'm not expecting to need anything in future, having the config struct in there allows for avoiding having to do any change in future to the function interface

Reply:
Adding `cfg` here bloats the interface considerably (see comment above), especially compared to what it was. Let's try and get away without needing this.

Reply:
We still need the `cfg.Repository` value in here. I'll change to just the single value rather than the structu